src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisDTOService.java

```diff
@@ -0,0 +1,160 @@
+import de.tum.in.www1.artemis.domain.iris.message.IrisMessage;
+import de.tum.in.www1.artemis.domain.iris.message.IrisTextMessageContent;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.service.RepositoryService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisBuildLogEntryDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisFeedbackDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisJsonMessageContentDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisMessageContentDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisMessageDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisProgrammingExerciseDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisResultDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisSubmissionDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.data.PyrisTextMessageContentDTO;
+
+@Service
+@Profile("iris")
+public class PyrisDTOService {
+
+    private static final Logger log = LoggerFactory.getLogger(PyrisDTOService.class);
+
+    private final GitService gitService;
+
+    private final RepositoryService repositoryService;
+
+    public PyrisDTOService(GitService gitService, RepositoryService repositoryService) {
+        this.gitService = gitService;
+        this.repositoryService = repositoryService;
+    }
+
+    /**
+     * Convert a ProgrammingExercise to a PyrisProgrammingExerciseDTO
+     *
+     * @param exercise the exercise
+     * @return the PyrisProgrammingExerciseDTO
+     */
+    public PyrisProgrammingExerciseDTO toPyrisDTO(ProgrammingExercise exercise) {
+        var templateRepositoryContents = getRepository(exercise.getTemplateParticipation()).map(repositoryService::getFilesWithContent).orElse(Map.of());
+        var solutionRepositoryContents = getRepository(exercise.getSolutionParticipation()).map(repositoryService::getFilesWithContent).orElse(Map.of());
+        Optional<Repository> testRepo = Optional.empty();
+        try {
+            testRepo = Optional.ofNullable(gitService.getOrCheckoutRepository(exercise.getVcsTestRepositoryUri(), true));
+        }
+        catch (GitAPIException e) {
+            log.error("Could not fetch existing test repository", e);
+        }
+        var testsRepositoryContents = testRepo.map(repositoryService::getFilesWithContent).orElse(Map.of());
+
+        return new PyrisProgrammingExerciseDTO(exercise.getId(), exercise.getTitle(), exercise.getProgrammingLanguage(), templateRepositoryContents, solutionRepositoryContents,
+                testsRepositoryContents, exercise.getProblemStatement(), toInstant(exercise.getReleaseDate()), toInstant(exercise.getDueDate()));
+    }
+
+    /**
+     * Convert a ProgrammingSubmission to a PyrisSubmissionDTO
```

Comment:
please add more documentation here as well, in particular if git operations are executed